api/v1beta1/pachyderm_types.go

```diff
@@ -171,24 +173,36 @@ type PachdOptions struct {
 
 // PostgresOptions allows user to customize Postgresql
 type PostgresOptions struct {
-	Disable      bool                         `json:"disable,omitempty"`
+	// Set disabled to true if you choose to provide
+	// an external postgresql database
+	Disable bool `json:"disable,omitempty"`
+	// Storage class for the postgresql persistent storage
 	StorageClass string                       `json:"storageClass,omitempty"`
 	Service      ServiceOverrides             `json:"service,omitempty"`
 	Resources    *corev1.ResourceRequirements `json:"resources,omitempty"`
 }
 
 // PachdPostgresConfig
 type PachdPostgresConfig struct {
+	// Hostname opr address  of the postgresql host
 	// +kubebuilder:default:=postgres
 	Host string `json:"host,omitempty"`
+	// Postgresql port number.
+	// Defaults to 5432 when not set
 	// +kubebuilder:default:=5432
 	Port int32 `json:"port,omitempty"`
 	// +kubebuilder:default:=disable
 	SSL string `json:"ssl,omitempty"`
+	// Username to use to connect to the database.
+	// Defaults to pachyderm
 	// +kubebuilder:default:=pachyderm
 	User string `json:"user,omitempty"`
 	// Will be autogenerated if left empty
-	Password string `json:"password,omitempty"`
+	Password string `json:"-"`
```

Comment:
If `PasswordSecretName` is unset, will `Password` be used? IMO it would be nice to include a comment on how these fields interact (even if they don't) because the names are similar.